### ISSUE TITLE
"Restore" event emitted two time tandem !!!????????

### ISSUE THREAD

**Issue by javidoweb** [nwjs/nw.js#5468]
Hi there
When window is maximized and then i minimize it and everything is great.
but when i unminimize it "restore" event's functions run two time ..... 
I am testing it on every version of nw and windows 10 and windows XP .
its happen just when window is maximized and you minimize it.
I have wrote following for maximize button:
```javascript
      var nw = require('nw.gui');

          var win = nw.Window.get();

          win.isMaximized = false;

          win.isMinimized = false;

...

$$('.winmaxi').click(function(event) {

    if (win.isMaximized) win.unmaximize();

    else win.maximize();

});

win.on('maximize', function() {

    win.isMaximized = true;

});


win.on('minimize', function() {

    win.isMinimized = true;

});

win.on('restore', function() {

    if (!win.isMinimized) {

        win.isMaximized = false;

    } else {

        win.isMinimized = false;

    }

    $$('.page-content').append('<h1>' + win.isMaximized + '</h1>');

});
```

**Comment by ghostoy** [comment]
@javidoweb The sequence of dispatching `restore` event is fixed in 0.18.6 (see #5388). If you are restoring from `minimize` state to `maximize`, you will be getting `restore` -> `maximize`. The final event will be correct. You won't need to track the states in `restore` event.

**Comment by ghostoy** [comment]
And FYI, the it's not fixed in LTS (0.14.x) yet.

**Comment by ghostoy** [comment]
Closed as duplicate of #5388. Please reopen it if you see different.
